backend/wmg/data/validation/validation.py

```diff
@@ -0,0 +1,384 @@
+    bronchus_serous_ontology_id = validation_cell_types["serous cell of epithelium of bronchus"]
+    secretory_cell_type_ids = [secretory_ontology_id, mucus_secretory_ontology_id, bronchus_serous_ontology_id]
+    AQP5_high_expression_cell_type_ids = goblet_cell_type_ids + secretory_cell_type_ids
+    AQP5_high_expression_cell_types = AQP5_human_lung_cube.query(
+        f"cell_type_ontology_term_id in {AQP5_high_expression_cell_type_ids}"
+    )
+    AQP5_non_high_expression_cell_types = AQP5_human_lung_cube.query(
+        f"cell_type_ontology_term_id not in {AQP5_high_expression_cell_type_ids}"
+    )
+
+    AQP5_high_expression_avg = (
+        AQP5_high_expression_cell_types.sum()["sum"] / AQP5_high_expression_cell_types.sum()["nnz"]
+    )
+    AQP5_non_high_expression_avg = (
+        AQP5_non_high_expression_cell_types.sum()["sum"] / AQP5_non_high_expression_cell_types.sum()["nnz"]
+    )
+    assert AQP5_high_expression_avg > AQP5_non_high_expression_avg
+
+
+def validate_expression_levels_for_particular_gene_dataset(path_to_expression_summary):
+    human_ont_id = validation_species_ontologies["human"]
+    human_lung_int = validation_tissues_with_many_cell_types["lung"]
+    MALAT1_ont_id = validation_gene_ontologies["MALAT1"]
+    CCL5_ont_id = validation_gene_ontologies["CCL5"]
+    with tiledb.open(path_to_expression_summary) as cube:
+        MALAT1_human_lung_cube = cube.df[
+            MALAT1_ont_id:MALAT1_ont_id, human_lung_int:human_lung_int, human_ont_id:human_ont_id
+        ]
+        CCL5_human_lung_cube = cube.df[
+            CCL5_ont_id:CCL5_ont_id, human_lung_int:human_lung_int, human_ont_id:human_ont_id
+        ]
+
+        MALAT1_expression = MALAT1_human_lung_cube.query(f"dataset_id == '{validation_dataset_uuid}'")
+        CCL5_expression = CCL5_human_lung_cube.query(f"dataset_id == '{validation_dataset_uuid}'")
+
+        malat1_expression_sum_by_cell_type = MALAT1_expression.groupby("cell_type_ontology_term_id").sum()["sum"]
+        ccl5_expression_sum_by_cell_type = CCL5_expression.groupby("cell_type_ontology_term_id").sum()["sum"]
+
+        expected_values = anndata.read_h5ad(
+            f"{pathlib.Path(__file__).parent.resolve()}/lung_map_3de0ad6d-4378-4f62-b37b-ec0b75a50d94.h5ad"
+        )
+        malat_expected = expected_values.obs.assign(MALAT1=expected_values.layers["rankit"].toarray()[:, 0])
+        ccl5_expected = expected_values.obs.assign(CCL5=expected_values.layers["rankit"].toarray()[:, 1])
+
+        expected_malat1_by_cell_type = malat_expected.groupby("cell_type_ontology_term_id").sum().MALAT1
+        expected_ccl5_by_cell_type = ccl5_expected.groupby("cell_type_ontology_term_id").sum().CCL5
+        # drop ccl5 cell types with expression value of zero (to match pipeline processing)
+        expected_ccl5_by_cell_type = expected_ccl5_by_cell_type[expected_ccl5_by_cell_type != 0]
+
+        # Todo actually compare once the rankit bug is fixed
```

Comment:
Already noted this dependency in that issue?

Reply:
Just merged that fix. I'm rerunning the pipeline now and I'll test that the arrays match/update the validation code to do an explicit check when its done